sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -108,10 +113,38 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
         "Not all required fields to update a received MSFAA record were provided.",
       );
     }
-
+    // Check if the msfaa number needs to be reactivated or simply updated with a signed date.
+    const retrievedMSFAARecord = await this.repo.findOne({
+      select: {
+        id: true,
+        studentId: true,
+        referenceApplicationId: true,
+        cancelledDate: true,
+        dateSigned: true,
+        offeringIntensity: true,
+      },
+      where: {
+        msfaaNumber,
```

Comment:
Should we not use the offering intensity in the where?

If there is any valid reason to not include the offering intensity in the criteria, a comment or TODO here will be clear.

Reply:
I guess, as they are using the findOne to fetch using the msfaa_number, might not have added the offering intensity.

Reply:
@guru-aot consider the example where we have a data  [MSFAANO1 - STUDENT1-PT] and [MSFAANO1-STUDENT1-FT]

and the file sent an MSFAA record [MSFAA01-STUDENT1-PT], what is the guarantee that this query will return with findone only  [MSFAANO1 - STUDENT1-PT]  ? it can return both. 
I believe we spoke about this in backlog grooming or dev chat but trying to check what is the way to move forward?

Reply:
MSFAA numbers are supposed to be unique even between full-time/part-time and they also start from different `seeds`. 
That said we never did additional checks around it, which does not mean that we should not have such additional constraints included (outside the PR).

Reply:
I have updated the query to include the `offeringIntensity` in the query.